### PR TITLE
update cache

### PR DESCRIPTION
**Pull request by ianpundar**
Updated score for inference.

Allow dataset to be cached in memory, and the training speed is doubled.

Dataset is also allowed to be cached to the disk, but this needs further testing.

### CODE REVIEW

File: tools/demo.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def visual(self, output, img_info, cls_conf=0.35):
         bboxes /= ratio
 
         cls = output[:, 6]
-        scores = output[:, 4] * output[:, 5]
+        scores = torch.sqrt(output[:, 4] * output[:, 5])
```
Comment (Joker316701882) [inline tools/demo.py:181]: This line shall be updated in a separate PR, since it serves different purpose with the other modifications.

File: tools/train.py [hunk 2]
```diff
@@ -66,11 +66,11 @@ def make_parser():
         help="Adopting mix precision training.",
     )
     parser.add_argument(
-        "--cache",
-        dest="cache",
-        default=False,
-        action="store_true",
-        help="Caching imgs to RAM for fast training.",
+        '--cache',
```
Comment (Joker316701882) [inline tools/train.py:69]: The quotation marks should better be consistent with others.

**Comment by FateScript** [comment]
LGTM. Thanks for your contribution @ianpundar :)
FYI @Joker316701882

**Comment by live-high** [comment]
seems it will cause oom when training with multiple processes
